org.mwc.debrief.track_shift/src/org/mwc/debrief/track_shift/controls/ZoneChart.java

```diff
@@ -447,29 +580,91 @@ else if (isResizeStart(z, event.x) || isResizeEnd(z, event.x))
 
         if (merge_1 != null && merge_2 != null && merge_1 != merge_2)
         {
+
           final Zone resize = merge_1.start < merge_2.start ? merge_1 : merge_2;
           final Zone delete = merge_1.start < merge_2.start ? merge_2 : merge_1;
+          final IntervalMarker deleteIntervalMarker = zoneMarkers.get(delete);
+          final IntervalMarker resizeIntervalMarker = zoneMarkers.get(resize);
           final XYPlot plot = (XYPlot) chart.getPlot();
-          {
-            final IntervalMarker intervalMarker = zoneMarkers.get(delete);
-            plot.removeDomainMarker(intervalMarker);
-            zoneMarkers.remove(delete);
-            zones.remove(delete);
-            fireZoneRemoved(delete);
-          }
 
-          if (resize.end < delete.end)
-            resize.end = delete.end;
+          final long endBefore = resize.end;
+          AbstractOperation mergeOp = new AbstractOperation("Merge Zone")
           {
 
-            final IntervalMarker intervalMarker = zoneMarkers.get(resize);
-            assert intervalMarker != null;
-            intervalMarker.setStartValue(resize.start);
-            intervalMarker.setEndValue(resize.end);
-          }
-          fireZoneResized(resize);
-          merge_1 = null;
-          merge_2 = null;
+            @Override
+            public IStatus execute(IProgressMonitor monitor, IAdaptable info)
+                throws ExecutionException
+            {
+
```

Comment:
@IanMayo ah I use this to enclose variable context so I do not have to always  come up with new variable names